### PR TITLE
Add data integrity proof

### PR DESCRIPTION
**Pull request by aljones15**
```js
{
  "@context": [
    "https://www.w3.org/2018/credentials/v1",
    {
      "ex": "https://example.org/examples#",
      "schema": "http://schema.org/",
      "rdf": "http://www.w3.org/1999/02/22-rdf-syntax-ns#",
      "BachelorDegree": "ex:BachelorDegree",
      "UniversityDegreeCredential": "ex:UniversityDegreeCredential",
      "degree": "ex:degree",
      "name": {
        "@id": "schema:name",
        "@type": "rdf:HTML"
      }
    },
    "https://w3id.org/security/data-integrity/v1"
  ],
  "id": "http://example.gov/credentials/3732",
  "type": [
    "VerifiableCredential",
    "UniversityDegreeCredential"
  ],
  "issuer": "did:key:z6Mkm7EEa7ZzhybKoA1EJm922PrvX9s74ZH4WF91pLH2Me7M",
  "issuanceDate": "2020-03-11T23:09:06.803Z",
  "credentialSubject": {
    "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
    "degree": {
      "type": "BachelorDegree",
      "name": "Bachelor of Science and Arts"
    }
  },
  "proof": {
    "type": "DataIntegrityProof",
    "created": "2022-09-14T20:41:59Z",
    "verificationMethod": "did:key:z6Mkm7EEa7ZzhybKoA1EJm922PrvX9s74ZH4WF91pLH2Me7M#z6Mkm7EEa7ZzhybKoA1EJm922PrvX9s74ZH4WF91pLH2Me7M",
    "cryptosuite": "eddsa-2022",
    "proofPurpose": "assertionMethod",
    "proofValue": "z4V1B8dYvmGjNjrt9CxbC7hR3pDgsg3FNSLk8hRKRgCiMokBe4g182vYhvP98DjFt8RsfDctD9VEbe5oqcq4L6QBk"
  }
}
```

To Do:
- [x] Fix the CI actions in node 18 for this.
  - [x] use a docker compose with the node and mongo images the ci uses
  - [x] try replicating locally with node 18 
  - [x] Release of this context: https://github.com/digitalbazaar/bedrock-multikey-context/pull/1/files
  - [x] Release of this context: https://github.com/digitalbazaar/bedrock-data-integrity-context/pull/1

### CODE REVIEW

File: lib/suites.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ */
+import {DataIntegrityProof} from '@digitalbazaar/data-integrity';
+import {Ed25519Signature2018} from '@digitalbazaar/ed25519-signature-2018';
+import {Ed25519Signature2020} from '@digitalbazaar/ed25519-signature-2020';
+import {
+  cryptosuite as eddsa2022CryptoSuite
+} from '@digitalbazaar/eddsa-2022-cryptosuite';
+
+const createEddsa2022Suite = async ({signer}) => {
+  // replace the milliseconds value with Z
```
Comment (dlongley) [inline lib/suites.js:12]:
```suggestion
  // remove milliseconds precision
```

File: lib/suites.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  // remove milliseconds precision
+  const date = new Date().toISOString().replace(/\.\d+Z$/, 'Z');
+  const cryptosuite = eddsa2022CryptoSuite;
+  return new DataIntegrityProof({signer, date, cryptosuite});
+};
+
+const SUPPORTED_SUITES = new Map([
+  ['Ed25519Signature2020', {
+    keyType: 'ed25519', createSuite:
+      ({signer}) => new Ed25519Signature2020({signer})
```
Comment (dlongley) [inline lib/suites.js:21]:
```suggestion
    keyType: 'ed25519',
    createSuite: ({signer}) => new Ed25519Signature2020({signer})
```

File: lib/suites.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  }],
+  ['Ed25519Signature2018', {
+    keyType: 'ed25519', createSuite:
+      ({signer}) => new Ed25519Signature2018({signer})
```
Comment (dlongley) [inline lib/suites.js:25]:
```suggestion
    keyType: 'ed25519',
    createSuite: ({signer}) => new Ed25519Signature2018({signer})
```

File: lib/suites.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import {
+  cryptosuite as eddsa2022CryptoSuite
+} from '@digitalbazaar/eddsa-2022-cryptosuite';
+
+const createEddsa2022Suite = async ({signer}) => {
+  // remove milliseconds precision
+  const date = new Date().toISOString().replace(/\.\d+Z$/, 'Z');
+  const cryptosuite = eddsa2022CryptoSuite;
+  return new DataIntegrityProof({signer, date, cryptosuite});
+};
```
Comment (dlongley) [inline lib/suites.js:16]: This should be moved to the bottom of the file and use this syntax:

```js
async function _createEddsa2022Suite({signer}) {
}
```

File: test/mocha/20-credentials.js [hunk 5]
```diff
@@ -23,674 +23,692 @@ const serviceType = 'vc-issuer';
...
-      } catch(e) {
-        error = e;
-      }
-      should.exist(error);
-      error.data.type.should.equal('DuplicateError');
+      });
     });
-  });
+  }
 });
```
Comment (dlongley) [inline test/mocha/20-credentials.js:714]: This is really hard to review ... I'm going to assume all that happened was that everything was indented here and put into a for loop for each suite. Is that right?

**Review by dlongley — CHANGES_REQUESTED** [review]

File: lib/suites.js (hunk 2 above)
Comment (aljones15) [reply lib/suites.js]: Addressed here: https://github.com/digitalbazaar/bedrock-vc-issuer/pull/97/commits/db785d4c9069d88cd817c383387762798033d2ba

File: lib/suites.js (hunk 3 above)
Comment (aljones15) [reply lib/suites.js]: Addressed here: https://github.com/digitalbazaar/bedrock-vc-issuer/pull/97/commits/db785d4c9069d88cd817c383387762798033d2ba

File: lib/suites.js (hunk 4 above)
Comment (aljones15) [reply lib/suites.js]: Agreed and address here: https://github.com/digitalbazaar/bedrock-vc-issuer/pull/97/commits/db785d4c9069d88cd817c383387762798033d2ba

File: test/mocha/20-credentials.js (hunk 5 above)
Comment (aljones15) [reply test/mocha/20-credentials.js:714]: Yes, we now test 3 suites: `Ed25519Signature2019`, `Ed25519Signature2020`, and `eddsa-2022`. All are passing in node 16.

File: lib/suites.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  const referenceId = `assertionMethod:${keyType}`;
+  const zcap = config.zcaps[referenceId];
+  if(!zcap) {
+    throw new Error(
+      `No capability available to sign using suite "${suiteName}".`);
+  }
+
+  return {zcap, createSuite, referenceId};
+}
+async function _createEddsa2022Suite({signer}) {
```
Comment (dlongley) [inline lib/suites.js:44]:
```suggestion
}

async function _createEddsa2022Suite({signer}) {
```

**Review by dlongley — APPROVED** [review]
One minor style nit -- let me know once that's done and I can merge and release this. Thanks!

**Pull request merged**